packages/headless/src/controllers/product-listing/range-facet/date-facet/headless-product-listing-date-filter.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  ConfigurationSection,
+  DateFacetSection,
+  SearchSection,
+} from '../../../../state/state-sections';
+import {loadReducerError} from '../../../../utils/errors';
+import {configuration, dateFacetSet, search} from '../../../../app/reducers';
+import {
+  logFacetClearAll,
+  logFacetSelect,
+} from '../../../../features/facets/facet-set/facet-set-analytics-actions';
+import {
+  buildCoreDateFilter,
+  DateFilter,
+  DateFilterInitialState,
+  DateFilterOptions,
+  DateFilterProps,
+  DateFilterRange,
+  DateFilterState,
+} from '../../../core/facets/range-facet/date-facet/headless-core-date-filter';
+import {ProductListingEngine} from '../../../../product-listing.index';
+import {fetchProductListing} from '../../../../features/product-listing/product-listing-actions';
+
+export {
+  DateFilterOptions,
+  DateFilterInitialState,
+  DateFilterRange,
+  DateFilterProps,
+  DateFilterState,
+  DateFilter,
+};
+
+export function buildDateFilter(
+  engine: ProductListingEngine,
+  props: DateFilterProps
+): DateFilter {
+  if (!loadDateFilterReducer(engine)) {
+    throw loadReducerError;
+  }
+
+  const coreController = buildCoreDateFilter(engine, props);
+  const {dispatch} = engine;
+  const getFacetId = () => coreController.state.facetId;
+
+  return {
+    ...coreController,
+    clear: () => {
+      coreController.clear();
+      dispatch(fetchProductListing());
+      dispatch(logFacetClearAll(getFacetId()));
+    },
+    setRange: (range) => {
+      const success = coreController.setRange(range);
+      if (success) {
+        dispatch(fetchProductListing());
+        dispatch(
+          logFacetSelect({
+            facetId: getFacetId(),
+            facetValue: `${range.start}..${range.end}`,
+          })
```

Comment:
I don't recall exactly why, but with `executeSearch` we went through quite a bit of trouble to make sure to log analytics after the state was updated with new results.

To do that here, you may be able to do:
```ts
dispatch(fetchProductListing()).then(() => logFacetSelect(...));
```